romancal/outlier_detection/tests/test_outlier_detection.py

```diff
@@ -1,86 +1,130 @@
+import os
+from pathlib import Path
+
+import numpy as np
 import pytest
+from astropy.units import Quantity
 
 from romancal.datamodels import ModelContainer
 from romancal.outlier_detection import OutlierDetectionStep, outlier_detection
 
 
+@pytest.fixture()
+def clean_up_after_test():
+    """
+    Clean up working directory after test.
+    """
+
+    def _clean_up_working_directory(pattern="*.asdf"):
+        """
+        Clean up the working directory by removing files matching the specified pattern.
+        Parameters
+        ----------
+        pattern : str, optional
+            The pattern of the file pattern to match (default is "*.asdf")
+
+        Returns
+        -------
+        None
+
+        """
+        for x in (Path(os.getcwd())).glob(pattern):
+            if x.exists():
+                x.unlink()
+
+    return _clean_up_working_directory
+
+
 @pytest.mark.parametrize(
     "input_models",
     [
         list(),
         [""],
         "",
         "testing",
-        {},
     ],
 )
 def test_outlier_raises_error_on_invalid_input_models(input_models):
-    """Test that OutlierDetection raises an error if input is not a ModelContainer."""
+    """Test that OutlierDetection raises an error if input cannot be parsed into a ModelContainer."""
 
     step = OutlierDetectionStep()
 
-    step.process(input_models)
+    with pytest.raises(Exception) as exec_info:
+        step(input_models)
```

Comment:
The code issues a warning and the test looks for an AttributeError (while actually a TypeError is raised).
This is a problem with the code and the test. 
Parsing invalid input into a `ModelContainer` raises a TypeError which is caught and a warning issued. So this test should be looking for and catching a warning.

The AttributeError which you see when you run the current code comes from `finalize_results`, not from parsing the input into a `ModelContainer`  and is raised because the input is not returned in that case (see my previous comment).